src/test/scala/org/openhorizon/exchangeapi/route/apikey/TestPostApiKeyRoute.scala

```diff
@@ -0,0 +1,408 @@
+package org.openhorizon.exchangeapi.route.apikey
+
+import org.json4s.DefaultFormats
+import org.json4s.jackson.JsonMethods
+import org.openhorizon.exchangeapi.auth.{Identity, Role}
+import org.openhorizon.exchangeapi.table.apikey.{ApiKeyMetadata, ApiKeyRow, ApiKeysTQ}
+import org.openhorizon.exchangeapi.table.organization.{OrgRow, OrgsTQ}
+import org.openhorizon.exchangeapi.table.user.{UserRow, UsersTQ}
+import org.openhorizon.exchangeapi.table.resourcechange.ResourceChangesTQ
+import org.openhorizon.exchangeapi.utility.{ApiUtils, Configuration, DatabaseConnection, ExchMsg}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.AnyFunSuite
+import scalaj.http.{Http, HttpResponse}
+import scala.concurrent.Await
+import org.apache.pekko.actor.ActorSystem
+import org.apache.pekko.event.LoggingAdapter
+import org.apache.pekko.http.scaladsl.model.{StatusCode, StatusCodes}
+import scala.concurrent.duration.{Duration, DurationInt}
+import slick.jdbc
+import slick.jdbc.PostgresProfile.api._
+import org.openhorizon.exchangeapi.auth.{Password, Role}
+import scala.concurrent.ExecutionContext.Implicits.global
+import _root_.org.openhorizon.exchangeapi.utility.{HttpCode,ApiTime}
+import java.sql.Timestamp
+import java.util.UUID
+
+class TestPostApiKeyRoute extends AnyFunSuite with BeforeAndAfterAll {
+  private val ACCEPT = ("Accept","application/json")
+  private val AWAITDURATION: Duration = 15.seconds
+  private val DBCONNECTION: jdbc.PostgresProfile.api.Database = DatabaseConnection.getDatabase
+  private val URL = sys.env.getOrElse("EXCHANGE_URL_ROOT", "http://localhost:8080") + "/v1/orgs/"
+  private val ROUTE = "/apikeys/"
+  private val PASSWORD = "password"
+  private implicit val formats: DefaultFormats.type = DefaultFormats
+  
```

Comment:
The fields that are not needed for testing you can leave blank in the actual DB. Save yourself grief/sanity. i.e. Email, description (where relevant), etc. That is for any test-suite.